bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/DateFormatExtension.java

```
@@ -0,0 +1,88 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ ******************************************************************************/
+
+package org.apache.sling.scripting.sightly.impl.engine.extension;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+import org.apache.sling.scripting.sightly.extension.RuntimeExtension;
+import org.apache.sling.scripting.sightly.render.RenderContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.osgi.service.component.annotations.Component;
+
+@Component(service = RuntimeExtension.class, property = { RuntimeExtension.NAME + "=" + "dateFormat" })
+/**
+ * Extension to format a date in HTL, use the following notation: ${ dateValue @ dateFormat}
+ * dateFormat used is from SimpleDateFormat
+ */
+public class DateFormatExtension implements RuntimeExtension {
+
+	private static final Logger LOG = LoggerFactory.getLogger(DateFormatExtension.class);
+
+	@Override
+	public Object call(final RenderContext renderContext, Object... arguments) {
+		ExtensionUtils.checkArgumentCount("dateFormat", arguments, 2);
+		Object dateValue = arguments[0];
+
+		String dateFormat = String.valueOf(arguments[1]);
+		
+		String returnValue = null;
+		
+		if (dateValue instanceof GregorianCalendar) {
+
+			GregorianCalendar gregorianCal = (GregorianCalendar) dateValue;
+			returnValue = format(gregorianCal.getTime(), dateFormat);
+
+		} else if (dateValue instanceof Calendar) {
+
+			Calendar cal = (Calendar) dateValue;
+			returnValue = format(cal.getTime(), dateFormat);
+
+		} else if (dateValue instanceof Date) {
+
+			Date dateAsDate = (Date) dateValue;
+			returnValue = format(dateAsDate, dateFormat);
+
+		}
```

Please throw an exception in case `dateValue` is of none of the three supported types.

---

done